src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/EETypeCreator.cs

```diff
@@ -565,7 +565,7 @@ private static unsafe int CreateArrayGCDesc(LowLevelList<bool> bitfield, int ran
             {
                 if (numSeries > 0)
                 {
-                    *ptr-- = (short)((bitfield.Count - last) * IntPtr.Size);
+                    *ptr-- = (short)((first + bitfield.Count - last) * IntPtr.Size);
```

Comment:
It looks like this code was broken for MD arrays for a long time. That means we did not have working tests sensitive to this for a while.

Here the original version . The `-2` there hardcodes the base size adjustment for SZ array. For MD array that would need to consider dimensions.

https://github.com/dotnet/runtime/blob/5b62e27e914a0067d85a24545b1013fa8b39b204/src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/EETypeCreator.cs#L568

https://github.com/dotnet/runtime/pull/86877  fixed this case partially, but broke the case where pointers are not at the start of the element. Thus this fix is needed.